gamemodes/cinema/gamemode/modules/theater/services/sh_hls.lua

```diff
@@ -12,7 +12,24 @@ function SERVICE:GetKey( url )
 	if string.sub( url.path, -5) == ".m3u8" then
 		return url.encoded
 	end
-	return false
+	match = string.match(url.encoded,"streamwat.ch/(%w+)/*$")
+	if match != nil and string.find(url.host,"streamwat.ch") then
+		http.Fetch("http://streamwat.ch/"..match.."/player.min.js",
```

Comment:
Won't work. http.Fetch is an asynchronous function, and the "returns" inside its callback will only return from the callback, not return something to the caller. GetKey is meant to be a lightweight function to identify a possibly matching URL anyway and it wouldn't be appropriate to use HTTP here. Let me update the repo with my youtube implementation so you can see how to get data asynchronously

Reply:
that's why I excluded the return at the end if the link is from streamwat.ch so it will let http.Fetch use the callback correctly

Reply:
I added the serverside of youtube and soundcloud. Youtube is an example of how to use the server to get extra data asychronously; you can use the .data field of the video to store anything arbitrary other than title/duration/thumbnail (which is why I renamed the identifier the "KEY", so I could use "DATA" as the name for the other stuff). The soundcloud serverside is an example of the plumbing needed to use the client for stuff

Reply:
That's not how it works... if the function "ends" without a "return" statement it implicitly returns nil. The callback part within http.Fetch won't be run until the webpage responds which would be like a second later, and by that time the entire context under which GetKey was called will be gone. If you checkout how youtube's serverside works, the function takes in "OnSuccess" and "OnFailure" arguments, both of which are functions to be run when the website has responded, which will continue the slow asychronous process of requesting the video.